src/View/WorkController.ts

```diff
@@ -45,7 +45,7 @@ export class WorkController {
     await Promise.all(
       poetTimestamps.map(timestamp =>
         this.collection.updateOne(
-          { 'timestamp.ipfsHash': timestamp.ipfsHash },
+          { 'timestamp.ipfsDirectoryHash': timestamp.ipfsHash },
```

Comment:
@kennylavender we have a bug here. With Claim Batching this function no longer works as expected. It should be updating all entries in the works collection with information on the timestamp. It's an upsert, so if we had no info on the claim yet, it would create an empty entry that only had the timestamp and later add claim info to it.

It now no longer works because it's creating a single entry for N claims. The timestamp also has a field named incorrectly now: `timestamp.ipfsHash` should now be `timestamp.ipfsDirectoryHash` I believe.

Since there's no 1-to-1 mapping any more, we'll need to figure out some logic to handle this correctly.

Reply:
I see, I thought I already checked for this but must have missed it. I think if it listens for the BatchReader instead of BlockchainReader it should solve it.

Reply:
Yup. It's not the easiest, either. I was thinking one way to handle this would be to add a new collection to view/api: something like `timestamps`. We store all the timestamp information there, and when new claims are added to the `works` collection, timestamp metadata is looked up in the `timestamps` collection and copied over. (We'd have two collections but API would still query one, without joins).

Copying data over from one collection to the other would require careful examination of the flows of the code. My gut feeling is it would be super easy to leave an undetected bug that generates invalid data over time.